Allow dry-run with our role

hush-hush opened the pull request:
- Fix the role to be compatible with ansible dry-run
- Fix downgrading from agent 5 to 6 on centos
- Improve our travis tests

hush-hush committed:
Fix the role to be compatible with ansible dry-run

hush-hush committed:
Fix downgrading from agent 5 to 6 on centos

hush-hush committed:
Improve our travis tests:

- Testing dry-run
- Testing installing agent 5 and 6
- Testing downgrading to agent 5
- Test the role with ansible 2.2 and 2.4

chaosoldier commented:
@hush-hush 
Thanks for fixing my issue. 
When is this PR merged?

hush-hush commented:
@chaosoldier I'm waiting for a review from @olivielpeau. We're currently focusing on releasing the next version of the agent (6.2.0). We will merge and release the ansible role as soon as possible,

chaosoldier commented:
@hush-hush
Thank you for replying. 

> We will merge and release the ansible role as soon as possible,

I understood!

olivielpeau commented on README.md:
```diff
@@ -41,7 +41,7 @@ To install agent5, you need to:
   (recommended) or leave it empty to always install the latest version (`5.*`).
 
-To downgrade from agent6 to agent5, you need to:
+To downgrade from agent6 to agent5, you need to (**on centos this will only work with ansible 2.4+**):
```
nit: `ansible 2.4 and up`?

olivielpeau replied:
you should also update the description of the `datadog_agent_allow_downgrade` variable in the `Role Variables` section of the readme

olivielpeau commented on tasks/pkg-debian.yml:
```diff
@@ -1,6 +1,7 @@
 ---
 - name: Install apt-transport-https
   apt: name=apt-transport-https state=present
+  when: not ansible_check_mode
```
why is this needed? Shouldn't the package be always in the apt repo?

hush-hush replied:
if no ansible job using apt was ever run, `--check` will fail because `python-apt` is not install.

```
fatal: [127.0.0.1]: FAILED! => {"changed": false, "msg": "python-apt must be installed to use check mode. If run normally this module can auto-install it."}
```

Adding the check allow users to check their playbook without installing python-apt.

olivielpeau commented on tasks/pkg-debian.yml:
```diff
@@ -18,18 +19,21 @@
   with_items:
     - "deb http://apt.datadoghq.com/ stable main"
     - "deb http://apt.datadoghq.com/ stable 6"
+  when: not ansible_check_mode
```
same question

olivielpeau commented on tasks/pkg-debian.yml:
```diff
@@ -18,18 +19,21 @@
     repo: "{{ datadog_apt_repo }}"
     state: "{% if datadog_agent5 %}absent{% else %}present{% endif %}"
     update_cache: yes
+  when: not ansible_check_mode
```
same question

olivielpeau reviewed (commented):
One general comment: according to https://docs.ansible.com/ansible/devel/user_guide/playbooks_checkmode.html#information-about-check-mode-in-variables, the variable `ansible_check_mode` is only available on versions >=2.1 of ansible. What happens on earlier versions of ansible?

Other than that and the inline nits, LGTM! The tests are very nice 👌

hush-hush commented:
@olivielpeau ansible 2.1 entered end of life last year (2017-06-01). I think we can no longer support it ?

olivielpeau reviewed (approved):
LGTM!

I'm fine with dropping support of <2.1 ansible versions, but let's not forget to update the declared minimum supported version of ansible here: https://github.com/DataDog/ansible-datadog/blob/2.0.3/meta/main.yml#L6 :)

hush-hush committed:
Drop support of EOL version of ansible

This allow us to use the latest feature from ansible like dry_run (ie: 'ansible_check_mode').
